src/util/file_bytes_stream.rs

```diff
@@ -53,3 +67,136 @@ impl FileBytesStream {
         Body::wrap_stream(self)
     }
 }
+
+#[derive(PartialEq, Eq)]
+enum FileSeekState {
+    NeedSeek,
+    Seeking,
+    Reading,
+}
+
+pub struct FileBytesStreamRange {
+    file_stream: FileBytesStream,
+    seek_state: FileSeekState,
+    start_offset: u64,
+}
+
+impl FileBytesStreamRange {
+    pub fn new(file: File, range: HttpRange) -> FileBytesStreamRange {
+        FileBytesStreamRange {
+            file_stream: FileBytesStream::new_with_limit(file, range.length),
+            start_offset: range.start,
+            seek_state: FileSeekState::NeedSeek,
+            
+        }
+    }
+}
+
+impl Stream for FileBytesStreamRange {
+    type Item = Result<Bytes, IoError>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
+        let Self {
+            ref mut file_stream,
+            ref mut seek_state,
+            start_offset,
+        } = *self;
+        if *seek_state == FileSeekState::NeedSeek {
```

Comment:
the difference between `NeedSeek` and `Seeking` is never encountered across polls, so we can eliminate this enum and convert it to a bool.

Reply:
Not true, we encounter this when the seek does not immediately resolve, the `NeedSeek` and `Seeking` states are required.